app/queries/articles/feeds/filter_query.rb

```diff
@@ -0,0 +1,92 @@
+module Articles
+  module Feeds
+    class FilterQuery
+      # [Ridhwana]: Important to note that this is a POC and is not thoroughly tested manually or
+      # in an automated fashion as yet. More verification is required.
+      MINIMUM_SCORE = -20
+
+      def self.call(...)
+        new(...).call
+      end
+
+      def initialize(feed_type: "explore", minimum_score: MINIMUM_SCORE,
+                     number_of_articles: ::Article::DEFAULT_FEED_PAGINATION_WINDOW_SIZE,
+                     page: 1, timeframe: nil, tag: nil, user: nil)
+        @filtered_articles = Article.all
+        @feed_type = feed_type
+        @minimum_score = minimum_score
+        @number_of_articles = number_of_articles
+        @page = page
+        @timeframe = timeframe
+        @tag = tag
+        @user = user
+      end
+
+      def call
+        if @tag.present?
+          @filtered_articles = filter_by_tags
```

Comment:
I'm not fan of a pattern where we assign a variable, then seemingly reassign it while doing some "magic" in the methods like `filter_by_tags`.
In my opinion, this seems counterintuitive: it seems like we replace the variable (`@filtered_articles`) value, but actually we modify it (we don't see it immediately and have to go through other methods to find it out)

What I suggest instead: pass a relation object. This way we will be more explicit: pass a relation, not modifying an instance variable "secretly".

E.g.
```
relation = Article.all

relation = filter_by_tags(relation) if @tag.present? # also use relation in the tags in all cases (for consistency), now it's not used in one of the branches and is replaced by `Tag#articles`
relation = filter_by_following_tags(relation) if @feed_type = following
```
example in the [article](https://mkdev.me/posts/how-to-use-query-objects-to-refactor-rails-sql-queries)


I would also prefer using private `attr_reader`s (or `attr_accessor`s if needed) for most instance variables, but maybe it's a question of preference.